astropy/units/equivalencies.py

```diff
@@ -811,3 +811,31 @@ def with_H0(H0=None):
     h100_val_unit = Unit(100/(H0.to_value((si.km/si.s)/astrophys.Mpc)) * astrophys.littleh)
 
     return Equivalency([(h100_val_unit, None)], "with_H0", kwargs={"H0": H0})
+
+
+def cosmological(z, cosmology=None, comoving=False):
+    """
+    Returns the equivalence between angles and cosmological distances.
+
+    Parameters
+    ----------
+    z : float
+        the redshift to consider for the conversion.
+    cosmology : `~astropy.cosmology.FLRW`
+        The cosmology to be used. If `None`, the astropy default cosmology.
+    comoving : bool
+        Use the comoving distances instead of proper distances
+
+    """
+    from astropy.cosmology import default_cosmology
+    if cosmology is None:
+        cosmology = default_cosmology.get()
+
+    if comoving:
+        return Equivalency([(si.arcsec, astrophys.kpc,
+                            lambda x: x / cosmology.arcsec_per_kpc_comoving(z),
```

Comment:
Separate from where this should live, there is no need to define simple multiplication/division function; instead do
```
Equivalency([(si.arcsec, u.Unit(astropys.kpc / cosmology.arcsec_per_kpc_comoving(z)))])
```

Reply:
See http://docs.astropy.org/en/latest/units/equivalencies.html#writing-new-equivalencies